appium-cookies/lib/cookies.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * derived from jQuery Cookie Plugin v1.4.1
+ * https://github.com/carhartl/jquery-cookies
+ */
+
+ //needed to communicate/translate between JSONWire cookies and regular JS cookies 
+
+import _ from 'lodash';
+
+export default class Cookies {
+
+  constructor (cookieString) {
+    this.cookieString = cookieString;
+  }
+
+  //parses the value if needed and converts the value if a converter is provided
+  convertCookie(value, converter) {
+    let parsedValue;
+
+    if (value.indexOf(`"`) === 0) {
+      value = value.slice(1, -1).replace(/\\"/g, `"`).replace(/\\\\/g, `\\`);
+    }
+    try {
+      parsedValue = decodeURIComponent(value.replace(/\+/g, ` `));
+    } catch (e) {}
+    
+    return _.isFunction(converter) ? converter(parsedValue) : parsedValue;
+  }
+
+  // takes arguments given and creates a JS Cookie 
+  writeCookie(key, value, options) { 
+    if (arguments.length > 1 && !_.isFunction(value)) {
```

Comment:
What is this check doing? Why is there a need to make sure that `value` is not a function?